url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/ScriptyEmbeddedRepl.java

```diff
@@ -0,0 +1,188 @@
+package com.sdicons.scripty;
+
+import com.sdicons.scripty.repl.ReplEngine;
+
+import java.io.*;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+@SuppressWarnings("UnusedDeclaration")
+public class ScriptyEmbeddedRepl
+extends ScriptyCapable
+{
+    public ScriptyEmbeddedRepl()
+    {
+    }
+
+    public ScriptyEmbeddedRepl(ScriptyCapable aScriptyFacade)
+    {
+        setReplEngine(aScriptyFacade.getReplEngine());
+    }
+
+    public void startLoop(final int aPort, final String aUid, final String aPwd)
+    {
+        final Thread lServerThread = new Thread()
+        {
+            @Override
+            public void run() 
+            {
+                final ReplServer lServer = new ReplServer(aPort, getReplEngine());
+                lServer.serveClients(aUid, aPwd);
+            }
+        };
+        lServerThread.setDaemon(true);
+        lServerThread.start();        
+    }
+}
+
+class ReplServer
+{
+    private ServerSocket server;
+    private ReplEngine scripty;
+
+    /**
+     * Construct the server, you have to call the serveClients method 
+     * for the server to start serving clients.
+     * @param aPort     The port where the server will listen for connections.
+     * @param aEngine   A REPL engine.
+     */
+    public ReplServer(int aPort, ReplEngine aEngine)
+    {
+        try
+        {
+            scripty = aEngine;
+
+            // Plain server.
+            server = new ServerSocket(aPort);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *server*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).